### PR TITLE
Implement CodePush Bundle Generation Command

### LINKED ISSUE #18: [Generalize][CLI] Implement CodePush Bundle Generation Command
1. Use the `react-native bundle` command to generate the JS bundle.  
2. Compile the JS bundle using the `hermesc` command.  
3. Create a CodePush bundle by zipping the files and generating a hash for identity verification.

- Provide a standalone command to generate bundles locally for testing.
- Output needed for implementing a release command.

### PR DESCRIPTION
**Pull request by floydkim**
closes #18 

The new code additions are substantial but largely derived from the `appcenter-cli` source code.
Moreover, as this code is already in use within the Soomgo app deployment scripts, its stability is not a concern.

Considering this, the parts of the code that require review are as follows:
- cli/commands/bundleCodePush.js
- cli/index.js

### Test Plan

1. Use verdaccio to publish the npm package to a local repository.
2. Install the package in the Soomgo ReactNative project.
3. In the project directory, run the command `npx code-push bundle` to confirm that the CodePush bundle is generated correctly.

image

The contents of the generated CodePush bundle were checked after unzipping, and it was confirmed that the bundle file is Hermes-compiled.
image

### CODE REVIEW

File: cli/functions/runHermesEmitBinaryCommand.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            shell.cp(source, destination);
+            shell.rm(source);
+            resolve();
+        } catch (e) {
+            reject(e);
+        }
+    }).then(() => {
+        if (!sourcemapOutput) {
+            // skip source map compose if source map is not enabled
+            return;
```
Comment (changwoolab) [inline cli/functions/runHermesEmitBinaryCommand.js:60]: I think using `async - await` is better than using `.then` in this case

File: cli/utils/promisfied-fs.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+async function walk(dir) {
+    const stats = await fs.stat(dir);
+    if (stats.isDirectory()) {
+        /**
+         * @type {string[]}
+         */
+        let files = [];
+        for (const file of await fs.readdir(dir)) {
+            files = files.concat(await walk(path.join(dir, file)));
```
Comment (changwoolab) [inline cli/utils/promisfied-fs.js:21]: As far as I know, `path.join`'s type is `join(...paths: string[]): string;`.

```suggestion
            files = files.concat(await walk(path.join(...dir, file)));
```

**Review by changwoolab — APPROVED** [review]
Great job 👏

File: cli/functions/runHermesEmitBinaryCommand.js (hunk 1 above)
Comment (floydkim) [reply cli/functions/runHermesEmitBinaryCommand.js:60]: @changwoolab 😃 
This code follows the original code of `appcenter-cli`.
It feels somewhat awkward to read. But, I prefer to leave this code as it is, given its stability, and focus resources on moving forward.

**Pull request merged**